### PR TITLE
Existing survey layout new

### PR DESCRIPTION
**Pull request by ranisa-gupta16**
## Description
image
Briefly describe the purpose of this pull request.

## How to Test

Please review the notebook screen which now shows records summary 
- Total records/sites
- My records/sites

naming of records/site being dynamically fetched from viewsets, default to record.

## Additional Information

Please ignore the table view as that has been fixed in another PR.

## Checklist

- [x] I have confirmed all commits have been signed.
- [x] I have added JSDoc style comments to any new functions or classes.
- [x] Relevant documentation such as READMEs, guides, and class comments are updated.

### CODE REVIEW

File: app/src/gui/components/notebook/index.tsx [hunk 1]
```diff
@@ -161,9 +163,22 @@ export default function NotebookComponent({project}: NotebookComponentProps) {
         .then(spec => {
           setUiSpec(spec);
           setViewsets(spec.viewsets);
+          // Set record label based on visible_types
+          if (spec.visible_types && spec.visible_types.length === 1) {
+            const recordType = spec.visible_types[0];
+            console.log('RRRRRRRR-111', recordType);
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/index.tsx:169]: Might want to clean these up?

File: app/src/gui/components/notebook/index.tsx [hunk 2]
```diff
@@ -138,6 +139,7 @@ export default function NotebookComponent({project}: NotebookComponentProps) {
   const theme = useTheme();
   const mq_above_md = useMediaQuery(theme.breakpoints.up('md'));
   const history = useNavigate();
+  const [recordLabel, setRecordLabel] = useState('RRecord');
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/index.tsx:142]: RRecord?

File: app/src/gui/components/notebook/index.tsx [hunk 3]
```diff
@@ -161,9 +163,22 @@ export default function NotebookComponent({project}: NotebookComponentProps) {
...
+          if (spec.visible_types && spec.visible_types.length === 1) {
+            const recordType = spec.visible_types[0];
+            console.log('RRRRRRRR-111', recordType);
+            const typeLabel = spec.viewsets[recordType]?.label || recordType;
+            console.log('RRRRRRRR-2222', typeLabel);
+
+            setRecordLabel(typeLabel);
+          } else {
+            setRecordLabel('Record');
+          }
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/index.tsx:176]: Looks like you can just derive the recordLabel from the viewsets - which already has a useState - so need for another useState for recordLabel. The component will already re-render when viewset is updated. Can be const in function scope.

File: app/src/gui/components/notebook/index.tsx [hunk 4]
```diff
@@ -161,9 +163,22 @@ export default function NotebookComponent({project}: NotebookComponentProps) {
...
           setLoading(false);
           setErr('');
         })
+
+        //todoranisa
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/index.tsx:181]: todo?

File: app/src/gui/components/notebook/record_summary.tsx
Comment (PeterBaker0) [inline app/src/gui/components/notebook/record_summary.tsx:1]: these are utility functions - so probably not `tsx` file - just `ts`, and can be put somewhere outside of GUI scope such as in utils?

File: app/src/gui/components/notebook/record_table.tsx [hunk 5]
```diff
@@ -358,16 +345,93 @@ function RecordsTable(props: RecordsTableProps) {
...
+        console.error('Error fetching user ID:', error);
+      }
+    };
+    fetchUser();
+  }, [project_id]);
+
+  useEffect(() => {
+    if (!rows || rows.length === 0 || !currentUser) {
+      return;
+    }
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/record_table.tsx:364]: If the rows length is zero, shouldn't it call the onRecordsCountChange(0)?

File: app/src/gui/components/notebook/record_table.tsx [hunk 6]
```diff
@@ -358,16 +345,93 @@ function RecordsTable(props: RecordsTableProps) {
...
+
+    const totalRecords = getTotalRecordCount(rows);
+
+    const myRecords = rows.filter(
+      record =>
+        getRowType({row: record} as GridCellParams) === 'Site' &&
+        record.created_by === currentUser
+    ).length;
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/record_table.tsx:372]: not using helper function? Maybe could make this a helper function or add optional filter of row type

File: app/src/gui/components/notebook/record_table.tsx [hunk 7]
```diff
@@ -358,16 +345,93 @@ function RecordsTable(props: RecordsTableProps) {
...
+    const fetchUser = async () => {
+      try {
+        const userId = await getCurrentUserId(project_id);
+        setCurrentUser(userId);
+      } catch (error) {
+        console.error('Error fetching user ID:', error);
+      }
+    };
+    fetchUser();
+  }, [project_id]);
```
Comment (PeterBaker0) [inline app/src/gui/components/notebook/record_table.tsx:359]: This pattern is best avoided 

a) we should have a context provider for current user
b) the useState -> useEffect combo for doing async is an anti-pattern in a codebase using React Query. I'd recommend a very simple custom hook useGetCurrentUser which takes project ID as prop, and returns the result of a useQuery doing what you've done above- then we get error, loading, caching etc all for free!

Too much to do context provider for this PR - so useQuery would be best.

**Review by PeterBaker0 — CHANGES_REQUESTED** [review]
Didn't review functionality - just code. Looks okay though I'm anxious about the abuse of useState/useEffect. In one instance it's not necessary i.e. you can just derive state from the other state which already force re-render (e.g. a computed/derived state). In the other, we are doing async fetches so best to use react query.

File: app/src/gui/components/notebook/index.tsx (hunk 2 above)
Comment (ranisa-gupta16) [reply app/src/gui/components/notebook/index.tsx:142]: Ah sorry, da,n
this was a test forgot to remove this. Thanks for picking.

File: app/src/gui/components/notebook/index.tsx (hunk 3 above)
Comment (ranisa-gupta16) [reply app/src/gui/components/notebook/index.tsx]: updated! :-)

File: app/src/gui/components/notebook/record_table.tsx (hunk 5 above)
Comment (ranisa-gupta16) [reply app/src/gui/components/notebook/record_table.tsx]: DOne

File: app/src/gui/components/notebook/record_table.tsx (hunk 6 above)
Comment (ranisa-gupta16) [reply app/src/gui/components/notebook/record_table.tsx]: code updated. 
:-)

File: app/src/gui/components/notebook/record_table.tsx (hunk 7 above)
Comment (ranisa-gupta16) [reply app/src/gui/components/notebook/record_table.tsx]: Yes ofcouse,
fixed this

**Comment by ranisa-gupta16** [comment]
Hey @PeterBaker0 , this PR has been updated based on above commnet. Please review and approve so it can be merged. :-) Thanks

**Review by PeterBaker0 — APPROVED** [review]
Made some minor changes

- added enter key handling on search input
- updated site counts to filter by the specified recordLabel on both counts so that buildings are not counted in total 
- changed "Total" to "Total {recordLabel}s" resulting in My Sites: and Total Sites:

The reason for this is that previously the total was including buildings, which made it feel like there were other people's sites, when actually these were your buildings.

**Review by stevecassidy — APPROVED** [review]
Added new config variable to the android build workflow. Otherwise looks ok.

**Pull request merged**